packages/features/schedules/lib/use-schedule/useTimesForSchedule.ts

```diff
@@ -1,24 +1,85 @@
+import { shallow } from "zustand/shallow";
+
 import dayjs from "@calcom/dayjs";
+import { useBookerStoreContext } from "@calcom/features/bookings/Booker/BookerStoreProvider";
+import type { BookerState } from "@calcom/features/bookings/Booker/types";
+import { getPrefetchMonthCount } from "@calcom/features/bookings/Booker/utils/getPrefetchMonthCount";
+import { isPrefetchNextMonthEnabled } from "@calcom/features/bookings/Booker/utils/isPrefetchNextMonthEnabled";
 
 import type { UseScheduleWithCacheArgs } from "./useSchedule";
 
 type UseTimesForScheduleProps = Pick<
   UseScheduleWithCacheArgs,
-  "month" | "monthCount" | "dayCount" | "selectedDate" | "prefetchNextMonth"
+  "month" | "dayCount" | "selectedDate" | "bookerLayout"
 >;
+
+interface UsePrefetchParams {
+  date: string;
+  month: string | null;
+  bookerLayout?: {
+    layout: string;
+    extraDays: number;
+    columnViewExtraDays: { current: number };
+  };
+  bookerState: BookerState;
+}
+
+const usePrefetch = ({ date, month, bookerLayout, bookerState }: UsePrefetchParams) => {
+  if (!bookerLayout) {
+    return null;
+  }
+
+  const dateMonth = dayjs(date).month();
+  const monthAfterAdding1Month = dayjs(date).add(1, "month").month();
+  const monthAfterAddingExtraDays = dayjs(date).add(bookerLayout.extraDays, "day").month();
+  const monthAfterAddingExtraDaysColumnView = dayjs(date)
+    .add(bookerLayout.columnViewExtraDays.current, "day")
+    .month();
+
+  const prefetchNextMonth = isPrefetchNextMonthEnabled(
+    bookerLayout.layout,
+    date,
+    dateMonth,
+    monthAfterAddingExtraDays,
+    monthAfterAddingExtraDaysColumnView,
+    month,
+    bookerLayout.extraDays
+  );
+
+  if (!prefetchNextMonth) {
+    return null;
+  }
+
+  const monthCount = getPrefetchMonthCount(
+    bookerLayout.layout,
+    bookerState,
+    monthAfterAdding1Month,
+    monthAfterAddingExtraDaysColumnView,
+    prefetchNextMonth
+  );
+
+  return { monthsToPrefetch: monthCount ?? 1 };
+};
+
 export const useTimesForSchedule = ({
-  month,
-  monthCount,
+  month: monthFromProps,
   selectedDate,
   dayCount,
-  prefetchNextMonth,
+  bookerLayout,
 }: UseTimesForScheduleProps): [string, string] => {
+  const [monthFromStore, bookerState] = useBookerStoreContext((state) => [state.month, state.state], shallow);
+  const month = monthFromStore ?? monthFromProps ?? null;
```

Comment:
Platform Component uses month from store already(see https://github.com/calcom/cal.com/blob/09-18-add-tests-_useprefetch/packages/platform/atoms/booker/BookerPlatformWrapper.tsx#L208)

To keep it consistent I am giving monthFromStore preference